src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -369,6 +369,34 @@ public void SetModelValue([NotNull] string key, [NotNull] ValueProviderResult va
             GetModelStateForKey(key).Value = value;
         }
 
+        /// <summary>
+        /// Clears <see cref="ModelStateDictionary"/> entries that match the key that is passed as parameter.
+        /// </summary>
+        /// <param name="key">The key of <see cref="ModelStateDictionary"/> to clear.</param>
+        public void ClearModelStateDictionaryEntries(string key)
```

Comment:
This should be more like `ClearValidationState()` It doesn't remove entries, it resets them. Also repeating `ModelStateDictionary` in the name is redundant, this is already part of that class.